modules/core/04-channel/keeper/upgrade.go

```diff
@@ -60,6 +60,7 @@ func (k Keeper) WriteUpgradeInitChannel(ctx sdk.Context, portID, channelID strin
 	k.Logger(ctx).Info("channel state updated", "port-id", portID, "channel-id", channelID, "previous-state", types.OPEN.String(), "new-state", types.INITUPGRADE.String())
 
 	emitChannelUpgradeInitEvent(ctx, portID, channelID, channel, upgrade)
+	return channel
```

Comment:
I am a bit of a `command-query separation` purist :) so I do like the convention that state-mutating functions should not return a value.

Reply:
I think we can definitely come back to this later, I do agree with @crodriguezvega's preferred approach, but I think for now its fine so that we can make progress quickly on the refactor.

Reply:
linking relevant issue opened for the try version #3825 (since I randomly bumped into it)